src/pages/Admin/Room/Card/RoomCard.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+
+import { RoomCardProps } from "./RoomCard.types";
+
+import * as S from "./RoomCard.styles";
+import Button from "components/atoms/Button";
+import { useSelector } from "hooks/useSelector";
+import VenueEventDetails from "pages/Admin/VenueEventDetails";
+import { Card } from "react-bootstrap";
+import { useUser } from "hooks/useUser";
+import { updateRoom } from "api/admin";
+import ToggleSwitch from "components/atoms/ToggleSwitch";
+
+const RoomCard: React.FC<RoomCardProps> = ({
+  room,
+  roomIndex,
+  venueId,
+  editHandler,
+  onEventHandler,
+}) => {
+  const { user } = useUser();
+  const events = useSelector((state) => state.firestore.ordered.events);
+  const filteredEvents =
```

Comment:
`useMemo`

Reply:
Also use selector function or 'wrapping hook' instead of the inline function passed to `useSelector` as per https://github.com/sparkletown/internal-sparkle-issues/issues/115